src/geometry.rs

```diff
@@ -1386,6 +1534,187 @@ mod tests {
 
         assert!(!geometry_with_orphans.has_no_orphan_normals());
     }
+
+    #[test]
+    fn test_geometry_mesh_genus_box_should_be_0() {
+        let geometry = cube_sharp_var_len([0.0, 0.0, 0.0], 1.0);
+        let edges: HashSet<UnorientedEdge> = geometry.unoriented_edges_iter().collect();
+
+        let genus = geometry.mesh_genus(&edges);
+        assert_eq!(genus, 0);
+    }
+
+    #[test]
+    fn test_geometry_mesh_genus_torus_should_be_1() {
+        let (faces, vertices) = torus();
+        let geometry = Geometry::from_triangle_faces_with_vertices_and_computed_normals(
+            faces.clone(),
+            vertices.clone(),
+            NormalStrategy::Sharp,
+        );
+        let edges: HashSet<UnorientedEdge> = geometry.unoriented_edges_iter().collect();
+
+        let genus = geometry.mesh_genus(&edges);
+        assert_eq!(genus, 1);
+    }
+
+    #[test]
+    fn test_geometry_mesh_genus_double_torus_should_be_2() {
+        let (faces, vertices) = double_torus();
+        let geometry = Geometry::from_triangle_faces_with_vertices_and_computed_normals(
+            faces.clone(),
+            vertices.clone(),
+            NormalStrategy::Sharp,
+        );
+        let edges: HashSet<UnorientedEdge> = geometry.unoriented_edges_iter().collect();
+
+        let genus = geometry.mesh_genus(&edges);
+        assert_eq!(genus, 2);
+    }
```

Comment:
Did all of these mesh genus tests come back alive? There's copies of them in `mesh_analysis.rs`

Also the one below, but the screen wasn't large enough for me to select them all 😂